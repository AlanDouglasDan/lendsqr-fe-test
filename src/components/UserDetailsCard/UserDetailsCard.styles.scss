@use "variables";

.information-card {
  background: variables.$white;
  border: 1px solid rgba(33, 63, 125, 0.06);
  box-shadow: 3px 5px 20px rgba(0, 0, 0, 0.04);
  border-radius: 4px;
  padding: 30px;

  .header16 {
    font-size: 16px;
    font-weight: 500;
  }

  .dark-blue {
    color: variables.$dark-blue;
  }

  .dim-grey {
    color: variables.$dim-grey;
  }

  .grid-5 {
    display: grid;
    grid-template-columns: repeat(5, 1fr);
    grid-gap: 30px;
  }

  .horizontal-line {
    border-bottom: 1px solid rgba(33, 63, 125, 0.1);
    margin: 30px 0;
  }
}
