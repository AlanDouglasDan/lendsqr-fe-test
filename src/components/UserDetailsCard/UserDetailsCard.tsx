import { UserInterface } from "types/user.types";
import "./UserDetailsCard.styles.scss";

type Props = {
  data: UserInterface;
};

const UserDetailsCard: React.FC<Props> = ({ data }) => {
  return (
    <div className="information-card">
      <div className="header16 dark-blue">Personal Information</div>

      <div className="grid-5 mt-30px">
        <div>
          <div className="text-uppercase dim-grey">full Name</div>
          <div className="header16 dim-grey mt-1">
            {data.profile.firstName} {data.profile.lastName}
          </div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Phone Number</div>
          <div className="header16 dim-grey mt-1">
            {data.profile.phoneNumber}
          </div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Email Address</div>
          <div className="header16 dim-grey mt-1">{data.email}</div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Bvn</div>
          <div className="header16 dim-grey mt-1">{data.profile.bvn}</div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Gender</div>
          <div className="header16 dim-grey mt-1">{data.profile.gender}</div>
        </div>

        <div>
          <div className="text-uppercase dim-grey mt-30px">Marital status</div>
          <div className="header16 dim-grey mt-1">Single</div>
        </div>

        <div>
          <div className="text-uppercase dim-grey mt-30px">Children</div>
          <div className="header16 dim-grey mt-1">None</div>
        </div>

        <div>
          <div className="text-uppercase dim-grey mt-30px">
            Type of residence
          </div>
          <div className="header16 dim-grey mt-1">Parent’s Apartment</div>
        </div>
      </div>

      <hr className="horizontal-line" />

      <div className="header16 dark-blue">Education and Employment</div>

      <div className="grid-4 m-0 mt-30px" style={{ gridGap: "30px" }}>
        <div>
          <div className="text-uppercase dim-grey">level of education</div>
          <div className="header16 dim-grey mt-1">{data.education.level}</div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">employment status</div>
          <div className="header16 dim-grey mt-1">
            {data.education.employmentStatus}
          </div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">sector of employment</div>
          <div className="header16 dim-grey mt-1">{data.education.sector}</div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Duration of employment</div>
          <div className="header16 dim-grey mt-1">
            {data.education.duration}
          </div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">office email</div>
          <div className="header16 dim-grey mt-1">
            {data.education.officeEmail}
          </div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Monthly income</div>
          <div className="header16 dim-grey mt-1">
            ₦{data.education.monthlyIncome[0]} - ₦
            {data.education.monthlyIncome[1]}
          </div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">loan repayment</div>
          <div className="header16 dim-grey mt-1">
            {data.education.loanRepayment}
          </div>
        </div>
      </div>

      <hr className="horizontal-line" />

      <div className="header16 dark-blue">Socials</div>

      <div className="grid-5 mt-30px">
        <div>
          <div className="text-uppercase dim-grey">Twitter</div>
          <div className="header16 dim-grey mt-1">{data.socials.twitter}</div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Facebook</div>
          <div className="header16 dim-grey mt-1">{data.socials.facebook}</div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Instagram</div>
          <div className="header16 dim-grey mt-1">{data.socials.instagram}</div>
        </div>
      </div>

      <hr className="horizontal-line" />

      <div className="header16 dark-blue">Guarantor</div>

      <div className="grid-5 mt-30px">
        <div>
          <div className="text-uppercase dim-grey">full Name</div>
          <div className="header16 dim-grey mt-1">
            {data.guarantor.firstName} {data.guarantor.lastName}
          </div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Phone Number</div>
          <div className="header16 dim-grey mt-1">
            {data.guarantor.phoneNumber}
          </div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Email Address</div>
          <div className="header16 dim-grey mt-1">grace@gmail.com</div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Relationship</div>
          <div className="header16 dim-grey mt-1">Sister</div>
        </div>
      </div>

      <hr className="horizontal-line" />

      <div className="header16 dark-blue opacity-0">Guarantor</div>

      <div className="grid-5 mt-30px">
        <div>
          <div className="text-uppercase dim-grey">full Name</div>
          <div className="header16 dim-grey mt-1">
            {data.guarantor.firstName} {data.guarantor.lastName}
          </div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Phone Number</div>
          <div className="header16 dim-grey mt-1">
            {data.guarantor.phoneNumber}
          </div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Email Address</div>
          <div className="header16 dim-grey mt-1">grace@gmail.com</div>
        </div>

        <div>
          <div className="text-uppercase dim-grey">Relationship</div>
          <div className="header16 dim-grey mt-1">Sister</div>
        </div>
      </div>
    </div>
  );
};

export default UserDetailsCard;
