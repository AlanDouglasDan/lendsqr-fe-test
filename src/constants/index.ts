import { UserInterface } from "../types/user.types";

enum API_URLS {
  GET_USERS = `https://6270020422c706a0ae70b72c.mockapi.io/lendsqr/api/v1/users`,
}

const TABLE_HEADERS: String[] = [
  "Organization",
  "Username",
  "Email",
  "Phone Number",
  "Date Joined",
  "Status",
];

const OPTIONS: Intl.DateTimeFormatOptions = {
  year: "numeric",
  month: "short",
  day: "numeric",
  hour: "numeric",
  minute: "numeric",
  hour12: true,
};

const DUMMY_USER_DATA: UserInterface = {
  id: "",
  createdAt: "",
  orgName: "",
  userName: "",
  email: "",
  phoneNumber: "",
  lastActiveDate: "",
  profile: {
    firstName: "",
    lastName: "",
    avatar: "",
    phoneNumber: "",
    gender: "",
    bvn: "",
    address: "",
    currency: "",
  },
  guarantor: {
    firstName: "",
    lastName: "",
    phoneNumber: "",
    gender: "",
    address: "",
  },
  accountBalance: "",
  accountNumber: "",
  socials: {
    facebook: "",
    twitter: "",
    instagram: "",
  },
  education: {
    level: "",
    employmentStatus: "",
    sector: "",
    duration: "",
    officeEmail: "",
    monthlyIncome: [],
    loanRepayment: "",
  },
};

const DUMMY_FILTER_DATA = {
  organization: "",
  username: "",
  email: "",
  phoneNumber: "",
  date: "",
  status: "",
};

const DUMMY_ORGANIZATION_DATA = [...Array(100)].map(
  (organization) => organization
);

export {
  API_URLS,
  TABLE_HEADERS,
  OPTIONS,
  DUMMY_USER_DATA,
  DUMMY_FILTER_DATA,
  DUMMY_ORGANIZATION_DATA,
};
