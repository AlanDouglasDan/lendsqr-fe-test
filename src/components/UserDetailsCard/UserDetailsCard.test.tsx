import { BrowserRouter } from "react-router-dom";
import { render } from "@testing-library/react";

import { DUMMY_USER_DATA } from "../../constants";
import UserDetailsCard from "./UserDetailsCard";

describe("User Details Card Component Test", () => {
  it("User Details Card Component Snapshot", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <UserDetailsCard data={DUMMY_USER_DATA} />
      </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
