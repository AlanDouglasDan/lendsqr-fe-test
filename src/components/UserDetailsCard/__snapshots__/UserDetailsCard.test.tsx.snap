// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Details Card Component Test User Details Card Component Snapshot 1`] = `
<DocumentFragment>
  <div
    class="information-card"
  >
    <div
      class="header16 dark-blue"
    >
      Personal Information
    </div>
    <div
      class="grid-5 mt-30px"
    >
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          full Name
        </div>
        <div
          class="header16 dim-grey mt-1"
        >
           
        </div>
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Phone Number
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Email Address
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Bvn
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Gender
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Marital status
        </div>
        <div
          class="header16 dim-grey mt-1"
        >
          Single
        </div>
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Children
        </div>
        <div
          class="header16 dim-grey mt-1"
        >
          None
        </div>
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Type of residence
        </div>
        <div
          class="header16 dim-grey mt-1"
        >
          Parent’s Apartment
        </div>
      </div>
    </div>
    <hr
      class="horizontal-line"
    />
    <div
      class="header16 dark-blue"
    >
      Education and Employment
    </div>
    <div
      class="grid-4 m-0 mt-30px"
    >
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          level of education
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          employment status
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          sector of employment
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Duration of employment
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          office email
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Monthly income
        </div>
        <div
          class="header16 dim-grey mt-1"
        >
          ₦ - ₦
        </div>
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          loan repayment
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
    </div>
    <hr
      class="horizontal-line"
    />
    <div
      class="header16 dark-blue"
    >
      Socials
    </div>
    <div
      class="grid-5 mt-30px"
    >
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Twitter
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Facebook
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Instagram
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
    </div>
    <hr
      class="horizontal-line"
    />
    <div
      class="header16 dark-blue"
    >
      Guarantor
    </div>
    <div
      class="grid-5 mt-30px"
    >
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          full Name
        </div>
        <div
          class="header16 dim-grey mt-1"
        >
           
        </div>
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Phone Number
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Email Address
        </div>
        <div
          class="header16 dim-grey mt-1"
        >
          grace@gmail.com
        </div>
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Relationship
        </div>
        <div
          class="header16 dim-grey mt-1"
        >
          Sister
        </div>
      </div>
    </div>
    <hr
      class="horizontal-line"
    />
    <div
      class="header16 dark-blue opacity-0"
    >
      Guarantor
    </div>
    <div
      class="grid-5 mt-30px"
    >
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          full Name
        </div>
        <div
          class="header16 dim-grey mt-1"
        >
           
        </div>
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Phone Number
        </div>
        <div
          class="header16 dim-grey mt-1"
        />
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Email Address
        </div>
        <div
          class="header16 dim-grey mt-1"
        >
          grace@gmail.com
        </div>
      </div>
      <div>
        <div
          class="text-uppercase dim-grey"
        >
          Relationship
        </div>
        <div
          class="header16 dim-grey mt-1"
        >
          Sister
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
